pkg/scheduler/framework/plugins/sameplacementaffinity/scoring_test.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package sameplacementaffinity
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/scheduler/framework"
+)
+
+func TestScore(t *testing.T) {
+	tests := []struct {
+		name             string
+		obsoleteBindings []*fleetv1beta1.ClusterResourceBinding
+		want             *framework.ClusterScore
+	}{
+		{
+			name: "has an obsolete binding",
+			obsoleteBindings: []*fleetv1beta1.ClusterResourceBinding{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "binding1",
+					},
+					Spec: fleetv1beta1.ResourceBindingSpec{
+						TargetCluster: clusterName,
+						State:         fleetv1beta1.BindingStateBound,
+					},
+				},
+			},
+			want: &framework.ClusterScore{ObsoletePlacementAffinityScore: 1},
+		},
+		{
+			name: "has an obsolete binding on other cluster",
+			obsoleteBindings: []*fleetv1beta1.ClusterResourceBinding{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "binding1",
+					},
+					Spec: fleetv1beta1.ResourceBindingSpec{
+						TargetCluster: "another-cluster",
+						State:         fleetv1beta1.BindingStateBound,
+					},
+				},
+			},
+			want: &framework.ClusterScore{ObsoletePlacementAffinityScore: 0},
+		},
+		{
+			name: "nil obsolete binding",
```

Comment:
A nit: it should be `no obsolete binding`?

Reply:
the array is nil and prefer to use "nil obsolete binding" to indicate that.